wolfcrypt/src/wc_pkcs11.c

```diff
@@ -544,10 +530,36 @@ int wc_Pkcs11Token_Init(Pkcs11Token* token, Pkcs11Dev* dev, int slotId,
                 }
             }
             if (ret == 0) {
-                if (slotCnt > 0)
+                if (tokenName != NULL && tokenSz > 0) {
+                    /* match based on token name */
+                    CK_TOKEN_INFO tinfo;
+                    ret = WC_INIT_E; /* default to token match not found */
+                    for (slotId = 0; slotId < (int)slotCnt; slotId++) {
+                        rv = dev->func->C_GetTokenInfo(slotId, &tinfo);
+                        PKCS11_RV("C_GetTokenInfo", rv);
+                        if (rv != CKR_OK) {
+                            ret = WC_HW_E;
```

Comment:
At some point, it might be good to have some specific error messages for PKCS#11 failures. It's currently not possible to tell what (and why it) failed without enabling debug prints. When running at customers, we can't just recompile and re-run. But this is probably material for a separate pull request / change.

Reply:
Thanks @fencekicker for the excellent feedback. I've made note of this and plan to followup with some improvements to the error codes in another PR.